ouroboros-network/src/Ouroboros/Network/PeerSelection/Governor.hs

```diff
@@ -537,8 +544,16 @@ peerSelectionGovernorLoop tracer
                             debugStateVar
                           }
                           jobPool
-                          pst = do
-    loop pst (Time 0) `catch` (\e -> traceWith tracer (TraceOutboundGovernorCriticalFailure e) >> throwIO e)
+                          pst
+                          readBigLedgerSnapshot
+                          ledgerPeersCtx =
+    handle (\e -> traceWith tracer (TraceOutboundGovernorCriticalFailure e) >> throwIO e) $ do
+      -- we run the verification job pre-emptively at startup because we may be stuck
+      -- in the TooOld ledger state before we catch up. Once ledger state turns YoungEnough,
+      -- this job ends. If for any reason ledger state turns TooOld from YoungEnough, this job
+      -- is restarted by the monitorLedgerStateJudgement job.
+      JobPool.forkJob jobPool (Monitor.jobVerifyPeerSnapshot readBigLedgerSnapshot ledgerPeersCtx)
```

Comment:
You don't need to start it preemptively, if the snapshot is invalid you want to shutdown diffusion so I don't think it makes sense for the governor to start without knowing if the snapshot is valid. I would synchronously verify the snapshot and only then start the loop.

Reply:
You should have a monitoring job then that is monitoring the change of the snapshot and runs the verification job asynchronously.